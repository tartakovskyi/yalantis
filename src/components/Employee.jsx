import React, { useContext, useState }  from 'react'
import { AppContext } from './App'


const Employee = ({ user:{ id, firstName, lastName }, letter}) => {

  const [active, setActive] = useState(0)
  const {changeUserStatus} = useContext(AppContext)

  const cls = active === 0 ? 'employee__name' : 'employee__name active'

  const onChangeValue = e => {
    const value = Number(e.target.value)
    setActive(value)
    changeUserStatus(id, letter, value)
  }

  return (
    <li className="employee">
      <h4 className={cls}>{firstName} {lastName}</h4>
      <form>
        <label className="form-check">
          <input 
            type="radio"
            name={`active_${id}`}
            value="0"
            id={`notActive_${id}`}
            className="form-check-input"
            onChange={onChangeValue}
            checked={active === 0}
          />
          <span className="form-check-label">not active</span>
        </label>
        <label className="form-check">  
          <input 
            type="radio"
            name={`active_${id}`}
            value="1"
            id={`active_${id}`}
            className="form-check-input"
            onChange={onChangeValue}
            checked={active === 1}
          />
          <span className="form-check-label">active</span>
        </label>
      </form>
    </li>
  )
}


export default Employee

