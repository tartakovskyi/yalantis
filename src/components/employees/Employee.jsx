import React, { useContext, useState } from "react";
import PropTypes from "prop-types";
import { AppContext } from "../App";

const Employee = ({ user: { id, firstName, lastName } }) => {
  const { activeUsers, changeUserStatus } = useContext(AppContext);
  const [active, setActive] = useState(activeUsers.includes(id) ? 1 : 0);

  const cls = active === 0 ? "employee__name" : "employee__name active";

  const onChangeValue = (e) => {
    const value = Number(e.target.value);
    setActive(value);
    changeUserStatus(id, value);
  };

  return (
    <li className="employee">
      <h4 className={cls}>
        {firstName} {lastName}
      </h4>
      <form>
        <label className="form-check">
          <input
            type="radio"
            name={`active_${id}`}
            value="0"
            id={`notActive_${id}`}
            className="form-check-input"
            onChange={onChangeValue}
            checked={active === 0}
          />
          <span className="form-check-label">not active</span>
        </label>
        <label className="form-check">
          <input
            type="radio"
            name={`active_${id}`}
            value="1"
            id={`active_${id}`}
            className="form-check-input"
            onChange={onChangeValue}
            checked={active === 1}
          />
          <span className="form-check-label">active</span>
        </label>
      </form>
    </li>
  );
};

Employee.propTypes = {
  user: PropTypes.shape({
    id: PropTypes.string.isRequired,
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired,
  }).isRequired,
};

export default Employee;
