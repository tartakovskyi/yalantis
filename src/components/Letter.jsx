import React  from 'react'
import Employee from './Employee'


const Letter = ({ letter, users }) => {

  const usersArr = Object.values(users).sort((a, b) => {
    if (a.lastName > b.lastName) {
      return 1
    } else if (a.lastName < b.lastName) {
      return -1
    } else return 0
  })

  return (
    <div className="col-4">
      <h3>{letter}</h3>
      {usersArr.length
        ?
        <ul className="users-list">
          {usersArr.map(user => <Employee user={user} letter={letter} key={user.id} />)}
        </ul>
        :
        <hr className="no-users" />
      }
    </div>
  )
}


export default Letter