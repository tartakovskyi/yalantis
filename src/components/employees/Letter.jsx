import React from "react";
import PropTypes from 'prop-types'
import Employee from "./Employee";

const Letter = ({ letter, users }) => {
  const usersArr = Object.values(users).sort((a, b) => {
    if (a.lastName > b.lastName) {
      return 1;
    } else if (a.lastName < b.lastName) {
      return -1;
    } else return 0;
  });

  return (
    <div className="col-4">
      <h3>{letter}</h3>
      {usersArr.length ? (
        <ul className="users-list">
          {usersArr.map((user) => (
            <Employee user={user} key={user.id} />
          ))}
        </ul>
      ) : (
        <hr className="no-users" />
      )}
    </div>
  );
};

Letter.propTypes = {
  letter: PropTypes.string.isRequired,
  users: PropTypes.object.isRequired
};

export default Letter;
