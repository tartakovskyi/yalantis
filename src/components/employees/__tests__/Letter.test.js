import React from "react";
import { render } from "@testing-library/react";
import Letter from "../Letter";
import Employee from "../Employee";

jest.mock("../Employee", () => () => <li class="employee">Employee</li>);

test("Letter should render correctly", () => {
  const letter = "H";
  const users = {
    "5e00928dba3201e530e3f9d1": {
      id: "5e00928dba3201e530e3f9d1",
      firstName: "Addie",
      lastName: "Harrington",
      dob: "2019-06-06T20:32:36.719Z",
    },
    "5e00928deed9b7022ce33839": {
      id: "5e00928deed9b7022ce33839",
      firstName: "Logan",
      lastName: "Holcomb",
      dob: "2019-09-07T08:33:06.470Z",
    },
    "5e00928d8b879ab59b26a59c": {
      id: "5e00928d8b879ab59b26a59c",
      firstName: "Medina",
      lastName: "Holmes",
      dob: "2019-09-24T00:47:33.894Z",
    },
  };

  const { container } = render(<Letter letter={letter} users={users} />);

  expect(container.firstChild).toMatchSnapshot();
  expect(container.querySelector(".employee").length).toBe(users.length);
});
