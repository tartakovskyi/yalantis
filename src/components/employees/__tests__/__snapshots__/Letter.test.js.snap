// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Letter should render correctly 1`] = `
<div
  class="col-4"
>
  <h3>
    H
  </h3>
  <ul
    class="users-list"
  >
    <li
      class="employee"
    >
      Employee
    </li>
    <li
      class="employee"
    >
      Employee
    </li>
    <li
      class="employee"
    >
      Employee
    </li>
  </ul>
</div>
`;
